Sources/Core/JavaIR.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  JavaIR.swift
+//  Core
+//
+//  Created by Rahul Malik on 1/4/18.
+//
+
+import Foundation
+
+public enum JavaVisibility: String {
+    case `public`
+    case protected
+    case `private`
+
+}
+
+/*
+
+ @AutoValue
+ public abstract class Animal {
+ public abstract String name();
+ public abstract int numberOfLegs();
+
+ public static Builder builder() {
+ return new AutoValue_Animal.Builder();
+ }
+
+ abstract Builder toBuilder();
+
+ public Animal withName(String name) {
+ return toBuilder().setName(name).build();
+ }
+
+ @AutoValue.Builder
+ public abstract static class Builder {
+ public abstract Builder setName(String value);
+ public abstract Builder setNumberOfLegs(int value);
+ public abstract Animal build();
+ }
+ }
+ */
+public struct JavaIR {
+
+    public struct Method {
+        let body: [String]
+        let signature: String
+
+        func render() -> [String] {
+            // HACK: We should actually have an enum / optionset that we can check for abstract, static, ...
+            if signature.contains("abstract") {
```

Comment:
Could be less hacky to include an enum in the `Method` struct that specified the "abstract" or not (and one for the other modifiers) rather than just doing a string contains (what if the method name is like `abstractBeanFactoryFactoryProxyBean` (this is Java after all)